[WIP] This changeset fixes the beesknees example plugin, which was br…

prinzdezibel opened the pull request:
…oken due

to refactoring. Things changed:
a) The navigation bar is now a React component
b) The default route permissions are no longer
maintained in Shop collection, but a dedicated Group collection

This closes the error reported here: https://forums.reactioncommerce.com/t/reaction-example-plugin-not-working/623

prinzdezibel committed:
[WIP] This changeset fixes the beesknees example plugin, which was broken due
to refactoring. Things changed:
a) The navigation bar is now a React component
b) The default route permissions are no longer
maintained in Shop collection, but a dedicated Group collection

This closes the error reported here: https://forums.reactioncommerce.com/t/reaction-example-plugin-not-working/623

machikoyasuda commented:
Welcome @prinzdezibel 🎊 

Your PR also closes https://github.com/reactioncommerce/reaction-example-plugin/issues/18

prinzdezibel committed:
[WIP] Refactor example-plugin for ReactionCommerce

Changes:
1) Replace footer and header templates with React components that are
sourced dynamically from registry's layout definitions.
2) Static page 'AboutUs' is now a React component
3) Fix for static page /about which pulled in wrong layout because of
   non-matching workflow value in register.js

Note: This change depends on register.js changes in core to work
propertly. Also it requires to be the core's footer React component
to be existent and registered as `Footer`.

prinzdezibel commented:
Further Changes:
1) Replace footer and header templates with React components that are
sourced dynamically from registry's layout definitions.
2) Static page 'AboutUs' is now a React component
3) Fix for static page /about which pulled in wrong layout because of
   non-matching workflow value in register.js

Note: This change depends on register.js changes in core to work
propertly. Also it requires to be the core's footer React component
to be existent and registered as `Footer`.

mikemurray commented on client/static/aboutus.js:
```diff
@@ -0,0 +1,18 @@
+    return (
+      <div className="container">
+        <h1>About Us</h1>
+        <p>Bee's Knees is a company manufacturing quality infant clothing with an emphasis on organic materials, high quality
```
Strings within tags should be wrapped like so `{"my string"}`

prinzdezibel replied:
Done. What's the reasoning behind it? How to deal with multi-line strings?

mikemurray commented on client/components/footer.js:
```diff
@@ -0,0 +1,18 @@
+import { replaceComponent } from "/imports/plugins/core/components/lib";
+import { Reaction } from "/client/api";
+
+const FooterBeesKnees = () =>
```
When doing stateless components with arrow functions always wrap in either a `()` or `{}` depending on your needs.

prinzdezibel replied:
done, like this:
```
const FooterBeesKnees = () => (
```
is this what you meant?

mikemurray commented on client/templates/layouts/core.js:
```diff
@@ -12,45 +14,62 @@ class CoreLayoutBeesknees extends Component {
-          <Blaze template={layoutHeader} className="reaction-navigation-header" />
-        }
+
+        { headerComponent && React.createElement(headerComponent, {}) }
```
just a nit-picky-style-thing, but I would have extra spaces with `{}` unless the linter told me otherwise.

prinzdezibel replied:
you probably meant "I would have *no* extra spaces" , no?
At least I've read that in another comment, if I remember correctly..

mikemurray commented on client/templates/layouts/core.js:
```diff
@@ -12,45 +14,62 @@ class CoreLayoutBeesknees extends Component {
+    if (mainComponent) {
+      return React.createElement(mainComponent, {});
+    }
+    else if (Template[template]) {
```
`} else if {`

mikemurray commented on client/templates/layouts/core.js:
```diff
@@ -12,45 +14,62 @@ class CoreLayoutBeesknees extends Component {
+      return React.createElement(mainComponent, {});
+    }
+    else if (Template[template]) {
+      return <Blaze template={template} />;
```
Though not necessary, trying to be consistent.

```
return (
  <Blaze template={template} />
);
```

mikemurray reviewed (dismissed):
Overall good start, just added some suggestions.

prinzdezibel committed:
Fix checkout step review, which was still refering to cartSubTotals
Blaze template instead of new React component CartSubTotal.

brent-hoover commented on reaction-example-plugin.iml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Pretty sure this isn't supposed to be checked in. I think it's a Webstorm file?

prinzdezibel replied:
Done. going to create a new PR soon.

brent-hoover commented on server/init.js:
```diff
@@ -20,16 +20,12 @@ function modifyCheckoutWorkflow() {
-  Shops.update(shop._id, {
-    $addToSet: { defaultRole: "about" }
-  });
+function addRolesToGroups() {
```
Let's double check the docs because there may be a method now that does this

prinzdezibel replied:
done. going to create a new PR soon.

prinzdezibel commented:
Going to close this, as this PR is superseded through:

https://github.com/reactioncommerce/reaction-example-plugin/pull/23
